spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/QuantTensor.scala

```diff
@@ -0,0 +1,1667 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.tensor
+
+import breeze.linalg.{DenseMatrix, DenseVector}
+import com.intel.analytics.bigdl.bigquant.BigQuant
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+import java.nio.ByteBuffer
+import org.apache.spark.mllib.linalg
+import org.apache.spark.mllib.linalg.Matrix
+import scala.reflect.ClassTag
+
+@SerialVersionUID(- 1766499387282335147L)
+private[bigdl] class QuantTensor[@specialized(Float) T: ClassTag](
+  private[bigdl] var _size: Array[Int],
+  private[bigdl] var _stride: Array[Int],
+  var nDimension: Int)(implicit ev: TensorNumeric[T]) extends Tensor[T] {
+  @transient private var desc = 0L
+  private var interStorage: Array[Byte] = null
+  private var setFromOther: Boolean = false
+
+  private val errorString = s"QuantizeTensor doesn't support this operation now"
+
+  def setStorage(buffer: Array[Byte]): this.type = {
+    interStorage = buffer
+    this
+  }
+
+  def getStorage: Array[Byte] = {
+    interStorage
+  }
+
+  def setStorageInJni(ptr: Long): Unit = {
+    release()
+    setFromOther = false
+    desc = ptr
+  }
+
+  def getStorageInJni: Long = {
+    desc
+  }
+
+  def isInitialized: Boolean = {
```

Comment:
We need to make sure that the tensor is always be properly initialized in its constructor